src/tests/ftest/dfuse/daos_build.py

```diff
@@ -250,3 +250,116 @@ def run_build_test(self, cache_mode, intercept=False, run_on_vms=False):
                 self.fail('{} over dfuse with il in mode {}.\n'.format(fail_type, cache_mode))
             else:
                 self.fail('{} over dfuse in mode {}.\n'.format(fail_type, cache_mode))
+
+    def test_dfuse_daos_build_pil4dfs(self):
+        """This test builds DAOS on a dfuse filesystem.
+
+        Use cases:
+            Create Pool
+            Create Posix container
+            Mount dfuse
+            Checkout and build DAOS sources.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
```

Comment:
Does this really require a hardware cluster?  Does it do something that needs specific hardware (i.e. pmem, nvme, I/B networking, etc.) or would this run on a VM cluster equally well?

Reply:
The current daos build test is running on hardware cluster.

@ashleypittman Do you know why we run daos build on a hardware cluster instead of a vm cluster? Thank you!

Reply:
Sorry. My bad. I did not notice that "test_dfuse_daos_build_wt_il" uses "vm" instead of "hw". I will change it to vm. Thank you!

Reply:
There are several variants of this test and they all run on whatever system is quickest.  Most of them run on real hardware but the `ioil` version runs on vms.

Reply:
Thank you! I will change it to use "vm" to consume less CI resource.

Reply:
daos build with libpil4dfs takes ~4 hours to finish in hw. Limited by the resources in vm, it takes more than 10 hours in vm. We probably need to run it in hw.